tests/aws/services/lambda_/test_lambda_api.py

```diff
@@ -4102,6 +4103,72 @@ def test_create_url_config_custom_id_tag_invalid_id(
+
+        create_lambda_function(
+            func_name=function_name,
+            zip_file=zip_contents,
+            runtime=Runtime.nodejs20_x,
+            handler="lambda_handler.handler",
+            Tags={TAG_KEY_CUSTOM_URL: custom_id_value},
+        )
+
+        def _assert_create_function_url(qualifier: str | None, expected_url_id: str):
+            params = {"FunctionName": function_name, "AuthType": "NONE"}
+            if qualifier:
+                # Note: boto3 will throw an exception if the Qualifier parameter is None or ""
+                params["Qualifier"] = qualifier
+
+            aws_client.lambda_.get_waiter("function_updated_v2").wait(FunctionName=function_name)
+            url_config_created = aws_client.lambda_.create_function_url_config(**params)
+            assert f"://{expected_url_id}.lambda-url." in url_config_created["FunctionUrl"]
+
+        def _assert_create_aliased_function_url(fn_version: str, fn_alias: str):
+            aws_client.lambda_.create_alias(
+                FunctionName=function_name, FunctionVersion=fn_version, Name=fn_alias
+            )
+            _assert_create_function_url(fn_alias, f"{custom_id_value}-{fn_alias}")
+
+        # Publishes a new version and creates an aliased URL
+        version = aws_client.lambda_.update_function_code(
+            FunctionName=function_name, ZipFile=zip_contents, Publish=True
+        ).get("Version")
+        _assert_create_aliased_function_url(fn_version=version, fn_alias="v1")
+
+        # Alias the $LATEST version
+        _assert_create_aliased_function_url(fn_version="$LATEST", fn_alias="latest")
+
+        # Update the code, creating an unpublished version
+        aws_client.lambda_.update_function_code(FunctionName=function_name, ZipFile=zip_contents)
+
+        # Create a URL for an unpublished function
+        _assert_create_function_url(qualifier=None, expected_url_id=custom_id_value)
+
+        # Ensure that these compound url-id's are stored correctly
+        with pytest.raises(aws_client.lambda_.exceptions.ResourceConflictException) as ex:
+            aws_client.lambda_.create_function_url_config(
+                FunctionName=function_name, AuthType="NONE", Qualifier="v1"
+            )
+        assert ex.match("ResourceConflictException")
+
+        # Ensure that all aliased URLs can be correctly retrieved
+        for alias in ["v1", "latest"]:
+            function_url = aws_client.lambda_.get_function_url_config(
+                FunctionName=function_name, Qualifier=alias
+            ).get("FunctionUrl")
+            assert f"://{custom_id_value}-{alias}.lambda-url." in function_url
+
+        # Finally, check if the none-aliased URL can be retrieved
+        function_url = aws_client.lambda_.get_function_url_config(FunctionName=function_name).get(
+            "FunctionUrl"
+        )
+        assert f"://{custom_id_value}.lambda-url." in function_url
```

Comment:
I was actually also getting issues with invoking functions on my local. Assumed it had to do with my local setup.  Let me try again and update this thread.

Reply:
Ahh, I think I found the issue. I wasn't adding a `Qualifier` to the `add_permissions` request. Since this is an aliased URL, the permission needs to be more fine-grained:

```
     aws_client.lambda_.add_permission(
            FunctionName=function_name,
            StatementId="urlPermission",
            Action="lambda:InvokeFunctionUrl",
            Principal="*",
            FunctionUrlAuthType="NONE",
            Qualifier=alias_name, # <-- this kwarg
        )
```

Reply:
We should add a behavior test case to cover that scenario. Basically, testing the alias-based custom function URL analogous to the base custom_id feature here: `aws.services.lambda_.test_lambda.TestLambdaURL.test_lambda_url_invocation_custom_id`

Reply:
Added a `test_lambda_url_invocation_custom_id_aliased` test for this 👍